compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/Types.java

```diff
@@ -4691,7 +4691,7 @@ private BType getIntersection(IntersectionContext intersectionContext, BType lhs
                     return symTable.semanticError;
                 }
 
-                return ImmutableTypeCloner.getEffectiveImmutableType(null, this, bType,
+                return ImmutableTypeCloner.getEffectiveImmutableType(symTable.builtinPos, this, bType,
```

Comment:
This isn't the correct pos, right??

Reply:
Yes, I checked similar cases which have done the same thing.
At this point can we get the correct pos?

Reply:
We need to pass it in. In `org.wso2.ballerinalang.compiler.util.ImmutableTypeCloner#defineImmutableRecordType` for example, this pos is used. We would be creating the symbol with a pos from the wrong module if we do this.

Can you please create an issue for where this is used incorrectly in other places?

Reply:
@MaryamZi, Is that okay if we pass pos in `IntersectionContext`?

Reply:
@MaryamZi, Is that okay if we pass the pos via `IntersectionContext`?

Reply:
Please ignore my comment, we can get the pos for `BTypeSymbol`

Reply:
Fixed

Reply:
See https://github.com/ballerina-platform/ballerina-lang/pull/35188/files#r820394596.